lib/Platform/Unicode/PlatformUnicodeICU.cpp

```diff
@@ -32,16 +41,22 @@ int localeCompare(
   // in two different ways to compare as equal.
   ucol_setAttribute(coll, UCOL_NORMALIZATION_MODE, UCOL_ON, &err);
   assert(U_SUCCESS(err) && "failed to set collator attribute");
+  return std::unique_ptr<UCollator, UCollatorDeleter>(coll, UCollatorDeleter());
+}
+} // namespace
+
+int localeCompare(
+    llvh::ArrayRef<char16_t> left,
+    llvh::ArrayRef<char16_t> right) {
+  static auto coll = openUCollator();
 
   auto result = ucol_strcoll(
```

Comment:
Could we add a comment about the safety of calling `ucol_strcoll` on the same `UCollator` from multiple threads?

I believe it is safe given https://unicode-org.github.io/icu/userguide/icu/design.html#thread-safe-const-apis, but we should document it. (and perhaps use a `std::unique_ptr<const UCollator>` to make it obvious)

Reply:
Added a comment. 
I tried `std::unique_ptr<const UCollator>` but turns out `ucol_close` does not accept a `const UCollator*`. I could use const_cast and reading `ucol_close`'s implementation I could see that there will be no UB due to const cast but it is not guaranteed on the API.

So I came up with current implementation where `getUCollator()` returns a `const UCollator*`. This will ensure that all the APIs invoked on the returned pointer will be thread-safe.

getUCollator() also owns the lifetime of the `unique_ptr<UCollator>`.